packages/react-ui/src/features/flows/components/import-flow-dialog.tsx

```diff
@@ -8,20 +18,80 @@ import {
   DialogTitle,
   DialogFooter,
 } from '@/components/ui/dialog';
+import { Input } from '@/components/ui/input';
+import { INTERNAL_ERROR_TOAST, toast } from '@/components/ui/use-toast';
+import { authenticationSession } from '@/lib/authentication-session';
+
+const ImportFlowDialog = ({ children }: { children: React.ReactNode }) => {
+  const [file, setFile] = useState<File | null>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const { mutate: createFlow } = useMutation<
+    PopulatedFlow,
+    Error,
+    FlowTemplate
+  >({
+    mutationFn: async (template: FlowTemplate) => {
+      const newFlow = await flowsApi.create({
+        displayName: template.name,
+        projectId: authenticationSession.getProjectId(),
+      });
+      return await flowsApi.update(newFlow.id, {
+        type: FlowOperationType.IMPORT_FLOW,
+        request: {
+          displayName: template.name,
+          trigger: template.template.trigger,
+        },
+      });
+    },
+    onSuccess: (flow) => {
+      window.location.href = `/flows/${flow.id}`;
+    },
```

Comment:
useNavigation instead, this will do hard refresh where it slow and it's no longer single page application